src/benchmarks/micro/corefx/System.Memory/ReadOnlySequence.cs

```diff
@@ -0,0 +1,182 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Memory;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Diagnostics.Windows;
+using BenchmarkDotNet.Diagnostics.Windows.Configs;
+using MicroBenchmarks;
+using Microsoft.Diagnostics.Tracing.Parsers.Clr;
+
+namespace MicroBenchmarks.corefx
+{
+    internal class BufferSegment<T> : ReadOnlySequenceSegment<T>
+    {
+        public BufferSegment(System.ReadOnlyMemory<T> memory)
+        {
+            Memory = memory;
+        }
+
+        public BufferSegment<T> Append(System.ReadOnlyMemory<T> memory)
+        {
+            var segment = new BufferSegment<T>(memory)
+            {
+                RunningIndex = RunningIndex + Memory.Length
+            };
+            Next = segment;
+            return segment;
+        }
+    }
+
+    [BenchmarkCategory(Categories.CoreFX)]
+    public partial class ReadOnlySequenceBenchmarks
+    {
+        private ReadOnlySequence<byte> _sequence;
+        private ReadOnlySequence<byte> _multiSegmentSequence;
+        private System.SequencePosition _start;
+        private System.SequencePosition _end;
+        private System.SequencePosition _ms_start;
+        private System.SequencePosition _ms_end;
+
+        //private System.Buffers.ReadOnlySequence<byte> _sequence;
+
+        [GlobalSetup]
+        public void GlobalSetup()
+        {
+            System.Memory<byte> memory = new System.Memory<byte>(Enumerable.Repeat((byte)1, 10000).ToArray());
+            _sequence = new ReadOnlySequence<byte>(memory);
+            _start =_sequence.GetPosition(10);
+            _end =_sequence.GetPosition(9990);
+
+            BufferSegment<byte> firstSegment = new BufferSegment<byte>(memory.Slice(0, memory.Length / 2 ));
+            BufferSegment<byte> secondSegment = firstSegment.Append(memory.Slice(memory.Length / 2 , memory.Length / 2 ));
+            _multiSegmentSequence = new ReadOnlySequence<byte>(firstSegment, 0, secondSegment, firstSegment.Memory.Length);
+            _ms_start = _multiSegmentSequence.GetPosition(10);
+            _ms_end = _multiSegmentSequence.GetPosition(9990);
+        }
+
+        [Benchmark]
+        public void StartPosition()
+        {
+            _sequence.Slice(_start);
+        }
+
+        [Benchmark]
+        public void Start()
+        {
+            _sequence.Slice(0);
```

Comment:
Same here/elsewhere, return the result. Also I would pick an offset that is non-zero, like 1. (same elsewhere)